File: 30 - Find the Second Largest Number in an Array/solution.js
function findSecondLargest(arr) {
  let first = -Infinity;
  let second = -Infinity;

  for (let num of arr) {
    if (num > first) {
      [second, first] = [first, num];
    } else if (num > second && num != first) {
      second = num;
    }
  }

  return second === -Infinity ? null : second;
}

// Test Cases
console.log(findSecondLargest([5, 3, 9, 1, 9])); // 5

File: 30 - Find the Second Largest Number in an Array/solution.py
def find_second_largest(arr):
    first = second = -float("inf")

    for num in arr:
        if num > first:
            second, first = first, num
        elif num > second and num != first:
            second = num
      

    return second if second != -float("inf") else None


# Test Cases
print(find_second_largest([5, 3, 9, 1, 9]))  # 5